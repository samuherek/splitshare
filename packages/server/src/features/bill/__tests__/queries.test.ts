import gql from 'graphql-tag';
import createTestServer from '../../../test-utils/server';
import console = require('console');

describe('bill queries', () => {
  test('myBills: query should return basic bill with all model fields', async () => {
    const MY_BILLS = gql`
      {
        myBills {
          id
          name
          icon
          createdAt
          updatedAt
        }
      }
    `;

    const { query } = await createTestServer({
      user: { id: 1 },
      models: {
        Bill: {
          getUserBills: jest.fn(() => [
            {
              id: 1,
              name: 'bill name',
              icon: null,
              createdAt: new Date('2000-01-01'),
              updatedAt: new Date('2000-01-01'),
            },
          ]),
        },
      },
    });

    const res = await query({ query: MY_BILLS });

    expect(res).toMatchSnapshot();
  });

  test('myBills: query including "users" should return list of bill users', async () => {
    const MY_BILLS_USERS = gql`
      {
        myBills {
          users {
            id
            email
            displayName
            photoUrl
          }
        }
      }
    `;

    const { query } = await createTestServer({
      user: { id: 1 },
      models: {
        Bill: {
          getUserBills: jest.fn(() => [{}]),
          getBillUsers: jest.fn(() => [
            {
              id: 1,
              email: 'email@email.com',
              displayName: null,
              photoUrl: null,
            },
          ]),
        },
      },
    });

    const res = await query({ query: MY_BILLS_USERS });

    expect(res).toMatchSnapshot();
  });

  test('myBills: query including "invites" should return list of bill invites', async () => {
    const MY_BILLS_INVITES = gql`
      {
        myBills {
          invites {
            id
            email
            pending
            createdAt
            updatedAt
            deletedAt
          }
        }
      }
    `;

    const { query } = await createTestServer({
      user: { id: 1 },
      models: {
        Bill: {
          getUserBills: jest.fn(() => [{}]),
          getBillInvites: jest.fn(() => [
            {
              id: 1,
              email: 'email@email.com',
              pending: true,
              createdAt: new Date('2000-01-01'),
              updatedAt: new Date('2000-01-01'),
              deletedAt: null,
            },
          ]),
        },
      },
    });

    const res = await query({ query: MY_BILLS_INVITES });

    expect(res).toMatchSnapshot();
  });

  test('bill: should return a bill with all base values', async () => {
    const QUERY = gql`
      query Bill($id: ID!) {
        bill(id: $id) {
          id
          name
          icon
          createdAt
          updatedAt
        }
      }
    `;

    const { query } = await createTestServer({
      user: { id: 1 },
      models: {
        Bill: {
          getById: jest.fn((...args) => ({
            id: 1,
            name: 'bill name',
            icon: null,
            createdAt: new Date('2000-01-01'),
            updatedAt: new Date('2000-01-01'),
          })),
        },
      },
    });

    const res = await query({
      query: QUERY,
      variables: { id: 1 },
    });

    expect(res).toMatchSnapshot();
  });
});
