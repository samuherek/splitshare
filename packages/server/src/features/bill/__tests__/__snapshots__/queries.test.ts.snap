// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bill queries bill: should return a bill with all base values 1`] = `
Object {
  "data": Object {
    "bill": Object {
      "createdAt": "2000-01-01",
      "icon": null,
      "id": "1",
      "name": "bill name",
      "updatedAt": "2000-01-01T00:00:00.000Z",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`bill queries myBills: query including "invites" should return list of bill invites 1`] = `
Object {
  "data": Object {
    "myBills": Array [
      Object {
        "invites": Array [
          Object {
            "createdAt": "2000-01-01",
            "deletedAt": null,
            "email": "email@email.com",
            "id": "1",
            "pending": true,
            "updatedAt": "2000-01-01T00:00:00.000Z",
          },
        ],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`bill queries myBills: query including "users" should return list of bill users 1`] = `
Object {
  "data": Object {
    "myBills": Array [
      Object {
        "users": Array [
          Object {
            "displayName": null,
            "email": "email@email.com",
            "id": "1",
            "photoUrl": null,
          },
        ],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`bill queries myBills: query should return basic bill with all model fields 1`] = `
Object {
  "data": Object {
    "myBills": Array [
      Object {
        "createdAt": "2000-01-01",
        "icon": null,
        "id": "1",
        "name": "bill name",
        "updatedAt": "2000-01-01T00:00:00.000Z",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
